eth_utils/decorators.py

```diff
@@ -13,63 +15,65 @@
 
 
 class combomethod(object):
-    def __init__(self, method):
+    def __init__(self, method: Callable[..., Any]) -> None:
         self.method = method
 
-    def __get__(self, obj=None, objtype=None):
+    def __get__(self, obj: Any=None, objtype: Type=None) -> Any:
         @functools.wraps(self.method)
-        def _wrapper(*args, **kwargs):
+        # Strings types used b/c pypy3 throws SegmentationFault if nested fn hints are actual types
+        def _wrapper(*args: 'Iterable[Any]', **kwargs: 'Dict[str, Any]') -> Callable[..., Any]:
             if obj is not None:
                 return self.method(obj, *args, **kwargs)
             else:
                 return self.method(objtype, *args, **kwargs)
         return _wrapper
 
 
-def _has_one_val(*args, **kwargs):
-    vals = itertools.chain(args, kwargs.values())
+def _has_one_val(*args: Iterable[Any], **kwargs: Dict[str, Any]) -> bool:
+    # Ignored because mypy cannot infer arg 1 of "chain"
+    vals = itertools.chain(args, kwargs.values())  # type: ignore
     not_nones = list(filter(lambda val: val is not None, vals))
     return len(not_nones) == 1
 
 
-def _assert_one_val(*args, **kwargs):
+def _assert_one_val(*args: Iterable[Any], **kwargs: Dict[str, Any]) -> None:
```

Comment:
I think a lot of these could use generics to make sure the inner type doesn't get lost to `Any`